starknet_devnet/starknet_wrapper.py

```diff
@@ -87,26 +87,27 @@ async def get_state(self):
         return starknet.state
 
     async def __update_state(self):
-        if not self.config.lite_mode_block_hash:
-            previous_state = self.__current_carried_state
-            assert previous_state is not None
-            current_carried_state = (await self.get_state()).state
-            state = await self.get_state()
+        previous_state = self.__current_carried_state
+        assert previous_state is not None
+        current_carried_state = (await self.get_state()).state
+        state = await self.get_state()
 
-            current_carried_state.block_info = self.block_info_generator.next_block(
-                block_info=current_carried_state.block_info,
-                general_config=state.general_config
-            )
+        current_carried_state.block_info = self.block_info_generator.next_block(
+            block_info=current_carried_state.block_info,
+            general_config=state.general_config
+        )
 
-            updated_shared_state = await current_carried_state.shared_state.apply_state_updates(
-                ffc=current_carried_state.ffc,
-                previous_carried_state=previous_state,
-                current_carried_state=current_carried_state
-            )
-            self.__starknet.state.state.shared_state = updated_shared_state
-            await self.__preserve_current_state(self.__starknet.state.state)
+        updated_shared_state = await current_carried_state.shared_state.apply_state_updates(
+            ffc=current_carried_state.ffc,
+            previous_carried_state=previous_state,
+            current_carried_state=current_carried_state
+        )
+
+        if not self.config.lite_mode_block_hash:
+            state.state.shared_state = updated_shared_state
+            await self.__preserve_current_state(state.state)
```

Comment:
Why this shouldn't be executed in the lite mode?

Reply:
It's the heaviest part of the method, as noted in the PR description. In lite mode, without it, timestamp manipulation is still possible, but getting real block hash and state updates is not.

Reply:
What do you mean by heaviest? Why is it needed for the regular workflow and not lite?

Reply:
Heaviest time-wise (as noted, takes around 89% of the whole method time). So with only skipping that in lite-mode, we still get the time benefit. And in regular mode it's needed for state update calculation and block state_root calculation.

Reply:
Maybe also `return generate_state_update` part can be indented (put under the if part, as it currently is on the master branch)

Reply:
This is not evident from the code so we should add a comment which explains it.

We should refactor this in the future so that state update and block generation are clearer.

Reply:
https://github.com/Shard-Labs/starknet-devnet/pull/105/commits/b643769c7ebe4bf5639343625b7b2494722c55bf